### PR TITLE
ENH issue#11977 Added ValueError to Timestamp p…

### LINKED ISSUE #11977: ERR: better error message on invalid timestamp parsing
```
In [2]: pd.Timestamp('2010011')
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-2-ccfd37d6ef5d> in <module>()
----> 1 pd.Timestamp('2010011')

/Users/jreback/pandas/pandas/tslib.pyx in pandas.tslib.Timestamp.__new__ (pandas/tslib.c:8958)()
    296         cdef _Timestamp ts_base
    297 
--> 298         ts = convert_to_tsobject(ts_input, tz, unit)
    299 
    300         if ts.value == NPY_NAT:

/Users/jreback/pandas/pandas/tslib.pyx in pandas.tslib.convert_to_tsobject (pandas/tslib.c:24122)()
   1232 
   1233     if util.is_string_object(ts):
-> 1234         return convert_str_to_tsobject(ts, tz, unit)
   1235 
   1236     if ts is None or ts is NaT:

/Users/jreback/pandas/pandas/tslib.pyx in pandas.tslib.convert_str_to_tsobject (pandas/tslib.c:26110)()
   1361                 ts = parse_datetime_string(ts, dayfirst=dayfirst, yearfirst=yearfirst)
   1362             except Exception:
-> 1363                 raise ValueError
   1364 
   1365     return convert_to_tsobject(ts, tz, unit)

ValueError: 
```

we are should have a nice error message here. This is normally caught for example in `to_datetime`, but if its only a single `Timestamp` no message.

### PR DESCRIPTION
**Pull request by MattRijk**
closes #11977

### CODE REVIEW

**Comment by jreback** [comment]
pls don't keep opening and closing pull requests

simply push to the same one

**Comment by MattRijk** [comment]
Question. How to I embed a graph in an issue?

**Comment by TomAugspurger** [comment]
https://guides.github.com/features/mastering-markdown/ search for images.

**Comment by jreback** [comment]
drag and drop

File: pandas/tslib.pyx [hunk 1]
```diff
@@ -1360,7 +1360,7 @@ cpdef convert_str_to_tsobject(object ts, object tz, object unit,
             try:
                 ts = parse_datetime_string(ts, dayfirst=dayfirst, yearfirst=yearfirst)
             except Exception:
-                raise ValueError
+                raise ValueError("year is out of range")
```
Comment (jreback) [inline pandas/tslib.pyx]: just pass this exception thru.

**Comment by jreback** [comment]
ideally like some tests that ``assert_raises_regex`` about the error message (add in ``tseries/test_timeseries``)

**Comment by jreback** [comment]
@MattRijk can you update according to comments

**Comment by jreback** [comment]
@MattRijk can you update

**Comment by jreback** [comment]
pls reopen if you'd like to update

**Comment by MattRijk** [comment]
@jreback. Sorry for not getting back to you. I worked on the update a couple of times for a while. I couldn't get the message to show correctly

**Pull request closed without merging**